sway-core/src/semantic_analysis/ast_node/expression/typed_expression/method_application.rs

```diff
@@ -375,10 +391,64 @@ fn find_method(
         }
         (ty, true) => ty.clone(),
     };
+    let ty = (|| -> CompileResult<TypeId> {
```

Comment:
Is it necessary to use a function closure here? This makes it harder to read and maintain IMO.